### PR TITLE
python-daemon 3.0.1

### PR DESCRIPTION
**Pull request by SC426**
**Destination channel:** defaults

### Links

- [PKG-4950](https://anaconda.atlassian.net/browse/PKG-4950)
- [Upstream repository](https://pagure.io/python-daemon)
- [Upstream changelog/diff](https://pagure.io/python-daemon/blob/efd2c25dcbca17725050337870d094544a6c8cf6/f/ChangeLog)
- Relevant dependency PRs:
  - https://github.com/AnacondaRecipes/luigi-feedstock/pull/2

### Explanation of changes:

- update version, sha256, and build script
- setuptools needs to be pinned to an older version or else pip install won't work https://answers.ros.org/question/362883/fail-building-foxy-modulenotfounderror-no-module-named-setuptoolsextern/

[PKG-4950]: https://anaconda.atlassian.net/browse/PKG-4950?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
-  sha256: 15c2c5e2cef563e0a5f98d542b77ba59337380b472975d2b2fd6b8c4d5cf46ca
+  sha256: 6c57452372f7eaff40934a1c03ad1826bf5e793558e87fef49131e6464b4dae5
 
 build:
   number: 0
-  script: {{ PYTHON }} -m pip install . --no-deps -vv
+  script: {{ PYTHON }} -m pip install . --no-deps -vv --no-build-isolation
   skip: True  # [win or (linux and s390x)]
 
 requirements:
```
Comment (lorepirri) [inline recipe/meta.yaml:17]: https://pagure.io/python-daemon/blob/main/f/setup.py#_67

```
setup_requires=[

  "docutils",
  "packaging",
  "setuptools",
  ],

 install_requires=[
   "setuptools >=62.4.0",
   "packaging",
   "lockfile >=0.10",
   ],
```

**Review by lorepirri — COMMENTED** [review]
some deps misplaced as far as I can see

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,30 +1,34 @@
...
   host:
     - python
     - pip
-    - setuptools
+    # Install won't work with setuptools >69.5.1
+    - setuptools 69.5.1
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - setuptools <=69.5.1
```
To make rebuilds easier.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -36,11 +40,15 @@ test:
 
 about:
   home: https://pagure.io/python-daemon/
+  description: |
+    This library implements the well-behaved daemon specification of
+    :pep:`3143`, “Standard daemon process library”.
   license: Apache-2.0
   license_family: APACHE
   license_file: LICENSE.ASF-2
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
  license: Apache-2.0 AND GPL-3.0-or-later
  license_family: Other
  license_file: 
    - LICENSE.ASF-2
    - LICENSE.GPL-3
```
The readme mentions both license being required.

File: recipe/meta.yaml (hunk 1 above)
Comment (ryanskeith) [reply recipe/meta.yaml:17]: This is interesting as well. https://pagure.io/python-daemon/issue/93

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,30 +1,34 @@
...
   host:
     - python
     - pip
-    - setuptools
+    # Install won't work with setuptools >69.5.1
+    - setuptools 69.5.1
     - wheel
-    - twine
+    - docutils
+    - packaging
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
```
It's not needed for v3.0.1 to build the package, see https://pagure.io/python-daemon/blob/067759fb4c802275d6d3b7dc34115ffb69bb79d5/f/setup.py#_62.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,30 +1,34 @@
...
   run:
     - python
     - lockfile >=0.10
     - setuptools
     - docutils
+    - packaging
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
```
It's not needed for v3.0.1 at runtime, see https://pagure.io/python-daemon/blob/067759fb4c802275d6d3b7dc34115ffb69bb79d5/f/setup.py#_64. It's not mentioned in any python modules

File: recipe/meta.yaml [hunk 6]
```diff
@@ -1,30 +1,34 @@
...
+    - setuptools 69.5.1
     - wheel
-    - twine
+    - docutils
+    - packaging
   run:
     - python
     - lockfile >=0.10
     - setuptools
     - docutils
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:30]:
```suggestion
```
docutils isn't needed at runtime. The maintainers try to do a trick https://pagure.io/python-daemon/blob/067759fb4c802275d6d3b7dc34115ffb69bb79d5/f/setup.py#_100 and https://pagure.io/python-daemon/blob/067759fb4c802275d6d3b7dc34115ffb69bb79d5/f/pyproject.toml#_13

File: recipe/meta.yaml [hunk 7]
```diff
@@ -36,11 +40,15 @@ test:
 
 about:
   home: https://pagure.io/python-daemon/
+  description: |
+    This library implements the well-behaved daemon specification of
+    :pep:`3143`, “Standard daemon process library”.
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    PEP 3143, “Standard daemon process library”.
```
to avoid any unexpected things on anaconda.org UI

**Comment by skupr-anaconda** [comment]
The PR title should be `python-daemon 3.0.1`, right?

File: recipe/meta.yaml (hunk 6 above)
Comment (SC426) [reply recipe/meta.yaml:30]: `+ pip check
python-daemon 3.0.1 requires docutils, which is not installed.`

Comment (SC426) [reply recipe/meta.yaml:30]: Pip check won't pass without this

File: recipe/meta.yaml (hunk 1 above)
Comment (lorepirri) [reply recipe/meta.yaml:17]: definitely

File: recipe/meta.yaml (hunk 6 above)
Comment (lorepirri) [reply recipe/meta.yaml:30]: you would need to patch it out

Comment (SC426) [reply recipe/meta.yaml:30]: Is that necessary in this situation?

File: recipe/meta.yaml (hunk 3 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: It is technically Apache-2.0 only if we only distribute the library without anything else.

**Review by skupr-anaconda — APPROVED** [review]

**Review by cbouss — APPROVED** [review]

**Pull request merged**